homeassistant/helpers/entity.py

```diff
@@ -698,10 +698,13 @@ def schedule_update_ha_state(self, force_refresh: bool = False) -> None:
         If state is changed more than once before the ha state change task has
         been executed, the intermediate state transitions will be missed.
         """
-        self.hass.create_task(
-            self.async_update_ha_state(force_refresh),
-            f"Entity {self.entity_id} schedule update ha state",
-        )
+        if force_refresh:
+            self.hass.create_task(
+                self.async_update_ha_state(force_refresh),
+                f"Entity {self.entity_id} schedule update ha state",
+            )
+        else:
+            self.hass.loop.call_soon_threadsafe(self.async_write_ha_state)
```

Comment:
Is there a reason we don't call `async_schedule_update_ha_state` and let it handle the two cases?

Reply:
There is not a real difference indeed, however, calling `async_schedule_update_ha_state` would create an additional task (which can be avoided/is not needed).

Reply:
Yes, for case 1, it would be two schedulings instead of one as done here. For case 2, it would be the same.

Reply:
Uh, no, if I call `async_schedule_update_ha_state` as suggested, it will always create an additional task... to call `async_schedule_update_ha_state` :)

Reply:
No, for case 2 we would schedule with `call_soon_threadsafe` once and then `async_schedule_update_ha_state` would write the state directly with `async_write_ha_state`. So it would be the same.